src/api/purchase/use-purchase.tsx

```diff
@@ -0,0 +1,47 @@
+import type { UseMutationOptions } from '@tanstack/react-query';
+
+import { client } from '../common';
+import { parseAxiosError, useBaseMutation } from '../common/utils';
+import type { ShoppingCartResponse as Response } from '../types';
+
+type CreditCard = {
+  card_number: string;
+  exp_month: number;
+  exp_year: number;
+  cvc: string;
+};
+
+type ShippingAddres = {
+  city: string;
+  country: string;
+  line_1: string;
+  line_2: string;
+  postal_code: string;
+  state: string;
+};
+
+type OrderVariables = {
+  credit_card: CreditCard;
+  shipping_address: ShippingAddres;
+};
+
+type Order = {
+  order: OrderVariables;
+};
+
+export const purchaseOrder = async (order: Order): Promise<Response> => {
+  try {
+    const { data } = await client.post('/orders', order);
```

Comment:
maybe you could have a constant with the endpoints, `const PURCHASE_ENDPOINT = { orders: '/orders' }` on `purchase/constants.ts`

Reply:
Done